### PR TITLE
Move to new manifest api

### PR DESCRIPTION
**Pull request by liamawhite**
Resolves https://github.com/tetratelabs/getenvoy/issues/10. Depends on https://github.com/tetratelabs/getenvoy-package/pull/26.

Signed-off-by: Liam White <liam@tetrate.io>

### CODE REVIEW

File: pkg/binary/getenvoy/fetch.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	tarball := filepath.Join(dst, "envoy.tar.gz")
+	f, err := os.OpenFile(tarball, os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		return "", err
+	}
+	defer func() { _ = f.Close() }()
+
+	bar := progressbar.NewOptions64(resp.ContentLength, progressbar.OptionSetDescription("[Fetching Envoy]"))
+	out := io.MultiWriter(f, bar)
+	_, err = io.Copy(out, resp.Body)
```
Comment (taiki45) [inline pkg/binary/getenvoy/fetch.go]: Q: don't we check a reponse status code before using a response body?

File: pkg/binary/getenvoy/fetch.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			if _, err := io.Copy(envoy, f); err != nil {
+				return err
+			}
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	if !found {
+		return errors.New("unable to find Envoy binary in downloaded tarball")
```
Comment (taiki45) [inline pkg/binary/getenvoy/fetch.go]: nit: Outputting tarball string (path?) can be useful.

File: pkg/binary/getenvoy/fetch.go (hunk 1 above)
Comment (liamawhite) [reply pkg/binary/getenvoy/fetch.go]: yes, we should!

File: pkg/binary/getenvoy/fetch.go (hunk 2 above)
Comment (liamawhite) [reply pkg/binary/getenvoy/fetch.go]: I don't think it's useful in this case. The defer ensures that the tarball is cleaned up. The URL would be useful but isn't in scope.

I could wrap any errors returned by this function to include the URL though.

File: go.mod [hunk 3]
```diff
@@ -16,3 +17,5 @@ require (
 	github.com/tetratelabs/getenvoy-package v0.0.0-20190718134531-9487f25b3273
 	github.com/xi2/xz v0.0.0-20171230120015-48954b6210f8 // indirect
 )
+
+replace github.com/tetratelabs/getenvoy-package => ../getenvoy-package
```
Comment (taiki45) [inline go.mod]: Q: is this on purpose?

Comment (liamawhite) [reply go.mod]: Ah, nope!

This was so I could use my updates from the local file directory. It won't pass lint until I remove it so when the other PR is merged I'll fix.

**Pull request merged**